application/cms/page_service.py

```diff
@@ -63,6 +66,16 @@ def create_page(self, page_type, parent, data, created_by, data_source_forms, ve
             position=len([c for c in parent.children if c.latest]),
         )
 
+        measure = Measure(slug=page.slug, position=page.position)
+        measure.subtopics = [
+            Subtopic.query.filter_by(
+                topic_id=Topic.query.filter_by(slug=parent.parent.slug).one().id, slug=parent.slug
```

Comment:
I find the embedded query hard to read. Maybe separate out the `Topic` query and assign that to a variable first instead?

Reply:
Or refactor this to a model method?  `Subtopic.find_by_slug_and_topic_slug(slug, topic_slug)`?

Reply:
Sure thing